builtin/game/misc.lua

```diff
@@ -74,18 +74,37 @@ function core.after(time, func, ...)
 	}
 end
 
-function core.check_player_privs(name, privs)
+function core.check_player_privs(player_or_name, ...)
+	local name = player_or_name
+	-- Check if we have been provided with a Player object.
+	if name.get_player_name ~= nil then
```

Comment:
Originally I thought about doing `type(name) == "table"`, but thought that checking for the field/function would be more explicit.

Should I change it to a test for a table?

Though, now that I say that, a `has_function(object, function_name)` function which would test `type(object) == "object" and type(object[function_name]) == "function"` might be cool for these moments. But that is something kinda differently.